usaspending_api/recipient/v2/views/list_recipients.py

```diff
@@ -1,59 +1,68 @@
-import logging
 import copy
+import logging
 
+from django.db.models import F, Q
 from rest_framework.response import Response
 from rest_framework.views import APIView
-from django.db.models import F, Q
 
 from usaspending_api.common.cache_decorator import cache_response
 from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
 from usaspending_api.common.validator.pagination import PAGINATION
 from usaspending_api.common.validator.tinyshield import TinyShield
 from usaspending_api.common.validator.utils import update_model_in_list
 from usaspending_api.recipient.models import RecipientProfile
-from usaspending_api.recipient.v2.lookups import SPECIAL_CASES
+from usaspending_api.recipient.v2.lookups import AWARD_TYPES, SPECIAL_CASES
 
 logger = logging.getLogger(__name__)
 
-
-# based from all_award_types_mappings in usaspending_api.awards.v2.lookups.lookups
-AWARD_TYPES = {
-    "contracts": {"amount": "last_12_contracts", "filter": "contract"},
-    "grants": {"amount": "last_12_grants", "filter": "grant"},
-    "direct_payments": {"amount": "last_12_direct_payments", "filter": "direct payment"},
-    "loans": {"amount": "last_12_loans", "filter": "loans"},
-    "other_financial_assistance": {"amount": "last_12_other", "filter": "other"},
-}
+award_types = list(AWARD_TYPES.keys()) + ["all"]
+RECIPIENT_MODELS = [
+    {"name": "keyword", "key": "keyword", "type": "text", "text_type": "search"},
+    {"name": "award_type", "key": "award_type", "type": "enum", "enum_values": award_types, "default": "all"},
+]
 
 
-def get_recipients(filters={}):
-    lower_limit = (filters["page"] - 1) * filters["limit"]
-    upper_limit = filters["page"] * filters["limit"]
-
+def build_duns_base_query(filters):
     qs_filter = Q()
     if "keyword" in filters:
         qs_filter |= Q(recipient_name__contains=filters["keyword"].upper())
         qs_filter |= Q(recipient_unique_id__contains=filters["keyword"])
 
+    if filters["award_type"] != "all":
+        qs_filter &= Q(award_types__overlap=[AWARD_TYPES[filters["award_type"]]["filter"]])
+
+    return qs_filter
+
+
+def get_recipients(filters={}, count=None):
+    lower_limit = (filters["page"] - 1) * filters["limit"]
+    upper_limit = filters["page"] * filters["limit"]
```

Comment:
Deriving these ... will TinyShield have enforced that the overall results to be returned for this page is less than the total (e.g. 1000)? If not, that may need to be enforced here so someone can't give values in their payload that derive a much wider page of results.

Reply:
Not quite sure what you're asking Keith. TinyShield prevents someone from requesting a limit greater than 1000. These `lower_limit` and `upper_limit` use that validated limit to determine the actual range of records to pull from the db.

Reply:
> TinyShield prevents someone from requesting a limit greater than 1000.
✅